src/jit/gentree.h

```diff
@@ -1747,6 +1753,15 @@ struct GenTree
     // sets "*pIsEntire" to true if this assignment writes the full width of the local.
     bool DefinesLocalAddr(Compiler* comp, unsigned width, GenTreeLclVarCommon** pLclVarTree, bool* pIsEntire);
 
+    regNumber RegAtDef() const
+    {
+        return GetRawRegNum();
+    }
+    regNumber RegAtUse() const
+    {
+        return isContained() ? REG_NA : GetRawRegNum();
+    }
+
```

Comment:
> In any case, I don't think this kind of logic belongs in GenTree because it is highly phase-dependent. Introducing this into GenTree in my opinion exacerbates existing problems with "containedness" and its phase-dependence.

Where would you put this logic? Pretty much all of the register-assignment-related APIs are already on `GenTree`.